### ISSUE TITLE
 Knex: Timeout acquiring a connection

### ISSUE THREAD

**Issue by rjeevaram** [louislam/uptime-kuma#4598]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

 I've been facing the below error in Uptimekuma Monitor recently. Has anyone faced this issue and any solution?
 
 **Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?**

### 📝 Error Message(s) or Log

 **Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?**

### 🐻 Uptime-Kuma Version

1.23.11

### 💻 Operating System and Arch

Running as K8s pods in  debian 11 Node

### 🌐 Browser

Chrome 122.0.6261.94

### 🖥️ Deployment Environment

- Runtime: 
- Database: 
- Filesystem used to store the database on: 
- number of monitors:

**Comment by CommanderStorm** [comment]
Please fill out the `Deployment Environment` to help us debug this.
In addition, what retention have you configured?

TL;DR:
In v1, reducing either retention/ping-rate/.. or moving to a faster storage solution is the only option to get around this problem.
In v2, we have likely resolved this problem. See https://github.com/louislam/uptime-kuma/issues/4500 for further details of what needs to happen before we can release this version.

**Comment by Lanhild** [comment]
Replying purely for metrics collection:

117 monitors
9.73 GB database

**Comment by CommanderStorm** [comment]
@Lanhild That is a record. Normally users run into problems at ~1.5GB. What kind of storage are you running this on?

**Comment by Lanhild** [comment]
@CommanderStorm I'm running it directly on Cloudron.

![image](https://github.com/louislam/uptime-kuma/assets/73610779/f9b13606-bf23-4e84-a33f-dd3234ba7d09)

**Comment by CommanderStorm** [comment]
I am going to close this as resolved by `v2.0` as I don't see evidence to the contrary.

> TL;DR:
> In v1, reducing either retention/ping-rate/.. or moving to a faster storage solution is the only option to get around this problem.
> In v2, we have likely resolved this problem. See #4500 for further details of what needs to happen before we can release this version.

**Comment by Jiriteach** [comment]
Ended up having to bin the db. SQLite db was over 6gb. Unfortunate but after starting again - no problems.

**Comment by Lanhild** [comment]
> @Lanhild That is a record. Normally users run into problems at ~1.5GB. What kind of storage are you running this on?

@CommanderStorm 

Culprit found. It's the `heartbeat` table that contains `68 439 519` rows.

**Comment by thebiblelover7** [comment]
@CommanderStorm Is the only option to wait for v2? I've set my activity history to 7 days, have only 12 monitors, I've wiped out the database and I still have the same issues....

**Comment by CommanderStorm** [comment]
Your issue seems somewhat strange.
- What storage are you running? (nfs, emmc, sd-card?) => 12 monitors with 7 days is way lower than reported by anybody..
- Could you have a look if this button works? (Was your db created before `v1.10`?)
  ![image](https://github.com/user-attachments/assets/b6fffb7f-da2b-4a46-96ae-786ec640d56c)

**Comment by thebiblelover7** [comment]
@CommanderStorm I'm running it on an Oracle Instance, so I don't know what storage they use.... But with other servers I've run on their instances, they seem to be running totally fine.

I've run Shrink Database multiple times even though I believe my db was created after v1.10

It has definitely been strange. Now I can't believe whenever I get spammed with 12 notifications saying all my services are down because of the above issue....

**Comment by CommanderStorm** [comment]
v2 will likely not solve your deployment issue. If you have suuuch slow IO, there is not much that can be done for you..
Maybe Oracle does have a plan or configuration where you get more than a few disk reads a second (Your IO must be [sloooooow](https://tenor.com/de/view/hello-old-people-computer-technology-biff-gif-12085559) to run into this that early)..

Maybe the machine you are running on has defective IO.. (contact support if benchmarks differ too much..)
Really can't tell without further debugging on your end.. ([IO-benchmarks](https://linuxreviews.org/HOWTO_Test_Disk_I/O_Performance), `lscpu`, a link to oracles page for your instance-type)

**Comment by thebiblelover7** [comment]
Well, @CommanderStorm , I recreated the instance, set the IO as fast as I could, and used ubuntu minimal instead of Oracle Linux... we'll see how it goes now. So far (last hour) it's working well.
